### PR TITLE
Feature/431/unit tests enhanced table row

### PR DESCRIPTION
**Pull request by tsivkadi**
## Ticket
[Feature/431](https://github.com/ita-social-projects/SpaceToStudy-Client/issues/431)

## Description
Implemented tests for EnhancedTableRow component

### CODE REVIEW

File: tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  beforeEach(() => {
+    useTableContext.mockReturnValue({
+      isSelection,
+      columns,
+      rowActions
+    })
+    render(<EnhancedTableRow isItemSelected={ false } item={ mockItem } refetchData={ refetchDataMock } />)
+  })
+
+  it('renders table row with all columns', async() => {
```
Comment (Yurenko) [inline tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx:50]: Delete 'async', it doesn't need here

Comment (tsivkadi) [reply tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx]: fixed

**Review by Yurenko — APPROVED** [review]

File: tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  })
+
+  it('renders table row with all columns', () => {
+    expect(screen.getByText('John Smith')).toBeInTheDocument()
+    expect(screen.getByText('john@email.com')).toBeInTheDocument()
+    expect(calculatedCellValueMock).toHaveBeenCalled()
+  })
+
+  it('renders table row with checkbox and menu icon', () => {
+    expect(screen.queryByRole('checkbox')).toBeInTheDocument()
```
Comment (Tolik170) [inline tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx:57]: To my mind this test case is redundant, you get checkbox and menu in next tests

File: tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    expect(screen.queryByTestId('menu-icon')).toBeInTheDocument()
+  })
+
+  it('calls handleSelectClick when checkbox is clicked', () => {
+    const checkbox = screen.getByRole('checkbox')
+    fireEvent.click(checkbox)
```
Comment (Tolik170) [inline tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx:63]: Add spaces between these 3 actions

File: tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    expect(handleSelectClickMock).toHaveBeenCalled()
+  })
+
+  it('renders action menu when menu icon is clicked', async() => {
+    const menuIcon = screen.getByTestId('menu-icon')
+    fireEvent.click(menuIcon)
+    expect(await screen.findByText('Delete')).toBeInTheDocument()
```
Comment (Tolik170) [inline tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx:70]: move **_(await screen.findByText('Delete')_**  to a varible and use it in expect as in tests above

File: tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  })
+
+  it('calls onAction function when clicking on the menu item', async() => {
+    const menuIcon = screen.getByTestId('menu-icon')
+    fireEvent.click(menuIcon)
+    await waitFor(() => {
+      fireEvent.click(screen.getByText('Delete'))
```
Comment (Tolik170) [inline tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx:77]: Try to use waitFor only for expect:
await waitFor(() => expect(rowActions[0].func).toHaveBeenCalledWith(mockItem._id))

File: tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+      expect(rowActions[0].func).toHaveBeenCalledWith(mockItem._id)
+    })
+  })
+  it('calls closeMenu function when clicking outside of the menu', async() => {
+    const menuIcon = screen.getByTestId('menu-icon')
+    fireEvent.click(menuIcon)
+    await waitFor(() => expect(screen.getByText('Delete')).toBeInTheDocument())
+    fireEvent.click(screen.getByRole('presentation').firstChild)
+    expect(menuIcon).toBeInTheDocument()
```
Comment (Tolik170) [inline tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx:86]: menuIcon was in the document from the beginning, not at all understand this expect

**Review by Tolik170 — CHANGES_REQUESTED** [review]

File: tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+      isSelection,
+      columns,
+      rowActions
+    })
+    render(<EnhancedTableRow isItemSelected={ false } item={ mockItem } refetchData={ refetchDataMock } />)
+  })
+
+  it('renders table row with all columns', () => {
+    expect(screen.getByText('John Smith')).toBeInTheDocument()
+    expect(screen.getByText('john@email.com')).toBeInTheDocument()
```
[the remark this answers is not in the document]
Comment (tsivkadi) [reply tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx]: fixed

File: tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx (hunk 6 above)
Comment (tsivkadi) [reply tests/unit/components/enhanced-table/enhanced-table-row/EnhancedTableRow.spec.jsx]: I changed this test in new commit

**Review by Tolik170 — APPROVED** [review]

**Review by MickhaelSh — APPROVED** [review]

**Review by maarekua — APPROVED** [review]

**Pull request merged**